CMakeLists.txt

```diff
@@ -132,6 +132,38 @@ if (CMAKE_BUILD_TYPE MATCHES "Coverage")
     COMMAND ${CMAKE_SOURCE_DIR}/scripts/record-coverage.sh ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
 endif()
 
+# Enable asserts for C/C++ code if requested
+#  from the LLVM project, https://opensource.apple.com/source/llvmCore/llvmCore-2358.3/CMakeLists.txt.auto.html
+if(CMAKE_BUILD_TYPE STREQUAL "Debug" )
+    option(PN_ENABLE_ASSERTIONS "Enable assertions" ON)
+else()
+    option(PN_ENABLE_ASSERTIONS "Enable assertions" OFF)
+endif()
+if(PN_ENABLE_ASSERTIONS)
+  # MSVC doesn't like _DEBUG on release builds.
+  if(NOT MSVC)
+    add_definitions(-D_DEBUG)
+  endif()
```

Comment:
This is not necessary at all - all we need to do is not define ``NDEBUG`` no need to define ``_DEBUG`` which does nothing for our code or any compiler library except MSVC.

Reply:
Ack, it does look useless. I only included it because thats what LLVM does, and it did not look harmful in any way.